code/controllers/subsystem/movement/zfall.dm

```diff
@@ -0,0 +1,37 @@
+/// Responsible for triggering mob zfalls when they are not already triggered by structure deconstruction or moving into openspace
+SUBSYSTEM_DEF(zfall)
+	name = "Zfall"
+	flags = SS_TICKER
+	wait = 5
+	runlevels = RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+	priority = FIRE_PRIORITY_ZFALL
+	var/list/atom/movable/openspace_inhabitants = list()
+
+/datum/controller/subsystem/zfall/proc/add_openspace_inhabitant(atom/movable/A)
+	if(!ismob(A) || (A in openspace_inhabitants))
+		return
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/remove_openspace_inhabitant, A)
+	openspace_inhabitants |= A
+
+/datum/controller/subsystem/zfall/proc/remove_openspace_inhabitant(atom/movable/A)
+	SIGNAL_HANDLER
+	if(!(A in openspace_inhabitants))
+		return
+	UnregisterSignal(A, COMSIG_PARENT_QDELETING)
+	openspace_inhabitants -= A
+
+/datum/controller/subsystem/zfall/fire(resumed)
+	var/list/removals = list()
+	for(var/atom/movable/A as() in openspace_inhabitants)
+		var/turf/T = get_turf(A)
+		if(!isopenspace(T))
+			removals |= A
+			continue
+		if(!A.zfalling)
+			T.zFall(A)
+		if(MC_TICK_CHECK)
+			return
```

Comment:
I have just noticed that this is built wrong and can't handle MC_TICK_CHECK properly.
See the other subsystems for how to properly handle enumerating over lists with MC_TICK_CHECKING. (You need to continue from where you left off rather than abandon everything at the end of the list)

Reply:
But it's not removing stuff from the list...  It needs to check everything on the tick 😔 I have no idea how to implement this properly